OneSignalSDK/onesignal/core/src/main/java/com/onesignal/internal/OneSignalImp.kt

```diff
@@ -405,17 +407,50 @@ internal class OneSignalImp : IOneSignal, IServiceProvider {
                 return
             }
 
+            // calling createAndSwitchToNewUser() replaces model with a default empty jwt
             createAndSwitchToNewUser()
-            operationRepo!!.enqueue(
-                LoginUserOperation(
-                    configModel!!.appId,
-                    identityModelStore!!.model.onesignalId,
-                    identityModelStore!!.model.externalId,
-                ),
-            )
 
-            // TODO: remove JWT Token for all future requests.
+            if (useIdentityVerification) {
+                // disable subscription if identity verification is on
+                operationRepo!!.enqueue(
+                    UpdateSubscriptionOperation(),
+                )
```

Comment:
This doesn't look like it disabled the subscription as the comments states, is this code correct?

Reply:
Addressed in the fixup commit